src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -545,7 +578,24 @@ public function register(Container $app)
             };
         });
 
-        $app['security.authentication_provider.dao._proto'] = $app->protect(function ($name) use ($app) {
+        $app['security.entry_point.guard._proto'] = $app->protect(function ($name, array $options) use ($app) {
+            if (isset($options['entry_point'])) {
+                // if it's configured explicitly, use it!
+                return $app[$options['entry_point']];
+            }
+            $authenticatorIds = $options['authenticators'];
+            if (count($authenticatorIds) == 1) {
+                // if there is only one authenticator, use that as the entry point
+                return $app[array_shift($authenticatorIds)];
```

Comment:
In practice, the way it's written should be fine, right? Specifically, `$authenticatorIds = $options['authenticators'];` makes a copy of the array, and then this new copy is only used for this one purpose.

But, +1 also for `reset()` as a solution.